internal/controller/siteconfig_controller.go

```diff
@@ -139,14 +141,25 @@ func (r *SiteConfigReconciler) Reconcile(ctx context.Context, req ctrl.Request)
 		return res, err
 	}
 
-	if !r.isSiteConfigValidated(siteConfig) {
-		if err := r.handleValidate(ctx, siteConfig); err != nil {
-			return requeueWithError(err)
-		}
+	// Validate SiteConfig
+	if err := r.handleValidate(ctx, siteConfig); err != nil {
+		return requeueWithError(err)
+	}
+
+	// Render, validate and apply templates
+	if rendered, err := r.handleRenderTemplates(ctx, siteConfig); err != nil {
+		return requeueWithError(err)
+	} else if !rendered {
+		return requeueImmediately(), nil
```

Comment:
In what conditions would we end up here?
Do we really want to continue to requeue constantly if the templates fail validation?

Wouldn't someone fixing the issue cause a new reconcile?

Reply:
Good point! I have removed the `requeueImmediately`.